src/core/client/admin/components/Comment/CommentContent.tsx

```diff
@@ -59,48 +19,15 @@ const CommentContent: FunctionComponent<Props> = ({
   children,
   highlight = false,
 }) => {
-  // Cache the expression used via memo. This will reduce duplicate renders of
-  // this comment content when the children change but the phrase configuration
-  // does not change. The regExp is already cached on a deeper level
-  // automatically, this is just lessening that impact further.
-  const expression = useMemo(() => {
-    // If we aren't in highlight mode for this comment, don't even attempt to
-    // generate the expression.
-    if (!highlight || !phrases) {
-      return null;
-    }
-
-    return getPhrasesRegExp(phrases);
-  }, [phrases, highlight]);
-
-  // Cache the parsed comment node. If the children cannot be parsed, this will
-  // be null.
-  const parsed = useMemo(() => {
-    if (typeof children !== "string") {
-      return null;
-    }
-
-    // Sanitize the input for display.
-    const node = getSanitize(highlight && !!phrases)(children);
-
-    // If the expression is available, then mark the nodes.
-    if (expression) {
-      markHTMLNode(node, expression);
-    }
-
-    return node;
-  }, [children, expression, highlight, phrases]);
-
-  if (parsed) {
-    return (
-      <div
-        className={cn(className, styles.root, highlight && styles.highlight)}
-        dangerouslySetInnerHTML={{ __html: parsed.innerHTML }}
-      />
-    );
-  }
-
-  return <>{children}</>;
+  return (
+    <CommentParser
+      phrases={phrases}
+      highlight={highlight}
+      className={cn(className, styles.root, highlight && styles.highlight)}
+    >
+      {children}
+    </CommentParser>
```

Comment:
I don't think we need `CommentParser` at all. It doesn't do anything different than `CommentContent` except it does not provide any default styles? Default styles can always be overwritten by passing a `className`.